BarrysPeanutsTemporalJava/src/main/java/barryspeanuts/ShoppingCartWorkflowImpl.java

```diff
@@ -0,0 +1,98 @@
+package barryspeanuts;
+
+import barryspeanuts.model.Address;
+import barryspeanuts.model.CreditCard;
+import barryspeanuts.model.PurchaseItem;
+import io.temporal.workflow.Workflow;
+import io.temporal.workflow.WorkflowQueue;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+
+public class ShoppingCartWorkflowImpl implements ShoppingCartWorkflow {
+  private static final Logger logger = Workflow.getLogger(ShoppingCartWorkflowImpl.class);
+
+  boolean exit = false;
+  List<PurchaseItem> purchaseItems = new ArrayList<>();
+  private final WorkflowQueue<Runnable> queue = Workflow.newWorkflowQueue(1024);
+
+  @Override
+  public void startWorkflow() {
+    logger.info("Starting Workflow for Barry's Peanuts");
+    Workflow.await(() -> exit);
+  }
+
+  @Override
+  public List<PurchaseItem> queryPurchaseItems() {
+    return this.purchaseItems;
+  }
+
+  @Override
+  public void addItem(PurchaseItem purchaseItem) {
+    this.purchaseItems.add(purchaseItem);
+  }
+
+  @Override
+  public void addItems(List<PurchaseItem> purchaseItems) {
+    this.purchaseItems.addAll(purchaseItems);
+  }
+
+  @Override
+  public void removeItem(PurchaseItem purchaseItem) {
+    this.purchaseItems.remove(purchaseItem);
+  }
+
+  @Override
+  public void checkOut() {
+    logger.info("Checking out {} purchase items.", this.purchaseItems.toArray().length);
+  }
+
+  @Override
+  public void pay(Address billingAddress, CreditCard creditCard) {
+    for (PurchaseItem purchaseItem : this.purchaseItems) {
+      purchaseItem.setBillingAddress(Optional.ofNullable(billingAddress));
+    }
+
+    JSONObject jsonObj = new JSONObject(billingAddress);
```

Comment:
Consider as `toString()` on `Address` if you need a formatted view of it

Reply:
This

```
logger.info(
        "Paying for purchase using credit card number {} for {} purchase item at billing address {}.",
        creditCard.getNumber(),
        this.purchaseItems.toArray().length,
            billingAddress.toString());
  }```

Produces this

```
Paying for purchase using credit card number 11112222333334444 for 3 purchase item at billing address barryspeanuts.model.Address@4d6e7d1a.
```

`Address.toString()` is what I used originally, but it didn't show the full structure of `Address, which is why I use `JSONObject`

Reply:
Right, I mean you would implement `toString()` on the model

Reply:
Addressed